pilot/pkg/bootstrap/namespacecontroller.go

```diff
@@ -70,50 +114,63 @@ func NewNamespaceController(ca *ca.IstioCA, core corev1.CoreV1Interface) (*Names
 				return core.Namespaces().Watch(options)
 			}}
 	})
-	c.namespaceStore, c.namespaceController =
-		cache.NewInformer(namespaceLW, &v1.Namespace{}, namespaceResyncPeriod, cache.ResourceEventHandlerFuncs{
-			UpdateFunc: c.namespaceUpdated,
-			AddFunc:    c.namespaceAdded,
+	_, c.namespaceController =
+		cache.NewInformer(namespaceLW, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				c.queue.Push(func() error {
+					return c.namespaceChange(newObj)
+				})
+			},
+			AddFunc: func(obj interface{}) {
+				c.queue.Push(func() error {
+					return c.namespaceChange(obj)
+				})
+			},
 		})
 	return c, nil
 }
 
-// When a namespace is created, Citadel adds its public CA certificate
-// to a well known configmap in the namespace.
-func (nc *NamespaceController) namespaceAdded(obj interface{}) {
+// Run starts the NamespaceController until a value is sent to stopCh.
+func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
+	go nc.namespaceController.Run(stopCh)
+	go nc.configMapController.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, nc.namespaceController.HasSynced, nc.configMapController.HasSynced)
+	log.Infof("Namespace controller started")
+	go nc.queue.Run(stopCh)
+}
+
+func (nc *NamespaceController) writeToNamespace(ns string) error {
+	meta := metav1.ObjectMeta{
+		Name:      CACertNamespaceConfigMap,
+		Namespace: ns,
+		Labels:    configMapLabel,
+	}
+	err := certutil.InsertDataToConfigMap(nc.core, meta, nc.data())
+	if err != nil {
+		return fmt.Errorf("error when inserting CA cert to configmap: %v", err)
+	}
```

Comment:
Return the error directly here? Otherwise it looks like the error message will repeat itself, e.g.

 `error when inserting CA cert to configmap: error when creating configmap %v: %v", meta.Name, err)`